### PR TITLE
add settings for Florincoin, enough to generate addresses with the right format

### PR DESCRIPTION
**Pull request by trentlarson**
Random thought: should I cut out more of the Flo*NetParams code if we don't know what it does?

Remaining steps:
- After merging, must publish to nexus before updating in other projects.

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/params/FloMainNetParams.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2013 Google Inc.
+ * Copyright 2015 Andreas Schildbach
```
Comment (stevesea) [inline core/src/main/java/org/bitcoinj/params/FloMainNetParams.java:3]: should this copyright include 2018 medici/trent?

File: core/src/main/java/org/bitcoinj/params/FloMainNetParams.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+public class FloMainNetParams extends AbstractBitcoinNetParams {
+    public static final int MAINNET_MAJORITY_WINDOW = 1000;
+    public static final int MAINNET_MAJORITY_REJECT_BLOCK_OUTDATED = 950;
+    public static final int MAINNET_MAJORITY_ENFORCE_BLOCK_UPGRADE = 750;
+
+    public FloMainNetParams() {
+        super();
+
+        /**
+           Begin section modified for FLO, believed to be correct.
```
Comment (stevesea) [inline core/src/main/java/org/bitcoinj/params/FloMainNetParams.java]: 'believed to be correct'? 

were these extracted from flo source?
are these values good for current FLO? Or 0.15? (or should it work for both)

**Review by stevesea — APPROVED** [review]

File: core/src/main/java/org/bitcoinj/params/FloMainNetParams.java (hunk 2 above)
Comment (cchrysostom) [reply core/src/main/java/org/bitcoinj/params/FloMainNetParams.java]: I think Trent's intent was only to get the addresses to work with importing a private key and signing data. My guess is all other parameters will have to be double-checked.

**Review by cchrysostom — COMMENTED** [review]
There are more things that need to be updated for Flo such as the DNS seeds, checkpoints, segwit, and so on. But, the changes did make it possible to generate Flo addresses. I'm okay as long as we take the time to verify all other parameters.

**Comment by cfox** [comment]
My only suggestion would be finding and extending relevant tests in https://github.com/OverstockMedici/bitcoinj/tree/master/core/src/test/java/org/bitcoinj/crypto (or wherever).

**Review by cfox — APPROVED** [review]
⚜️

File: core/src/main/java/org/bitcoinj/params/FloMainNetParams.java (hunk 2 above)
Comment (trentlarson) [reply core/src/main/java/org/bitcoinj/params/FloMainNetParams.java]: OK, I make some changes to the commentary.

Those values should be good for all versions of Florincoin.

**Comment by trentlarson** [comment]
@cfox Is absolutely right about the tests!  I'm gonna make a Jira task to fix those (in this sprint).

**Comment by cchrysostom** [comment]
For future changes needed for a full conversion to flo, take a look at the commit history at https://github.com/floblockchain/flo/commits/flo-master.

**Comment by trentlarson** [comment]
Unit test Jira issue: https://mediciventures.atlassian.net/browse/DS-63

**Pull request merged**